android/shared/src/main/kotlin/kmp/android/shared/navigation/Destination.kt

```diff
@@ -0,0 +1,149 @@
+package kmp.android.shared.navigation
+
+import android.net.Uri
+import androidx.compose.foundation.layout.ColumnScope
+import androidx.compose.runtime.Composable
+import androidx.navigation.NamedNavArgument
+import androidx.navigation.NavBackStackEntry
+import androidx.navigation.NavDeepLink
+import androidx.navigation.NavGraphBuilder
+import androidx.navigation.compose.composable
+import androidx.navigation.compose.dialog
+import com.google.accompanist.navigation.material.ExperimentalMaterialNavigationApi
+import com.google.accompanist.navigation.material.bottomSheet
+
+/**
+ * Base class for navigation graph root class.
+ */
+interface RootDestination {
+    val route: String
+}
+
+/**
+ * Base class for any destination the app uses.
+ */
+abstract class Destination(parent: RootDestination?) {
+    /**
+     * The root id of the destination graph used in the route <root>/<destinationId>/<argument>...
+     * In case the parent is null, we do not need any prefix and use the destinationId by itself <destinationId>/<argument>...
+     */
+    private val root: String = parent?.route?.let { "$it/" } ?: ""
+
+    /**
+     * The id of the destination, without root id and nav arguments.
+     */
+    protected abstract val destinationId: String
```

Comment:
The "ID" could be mistaken for XML nav graph resource ID. I've used "routeDefinition" in other projects to name the "raw" path and "route" as the path instance (the built URI)

Reply:
I like `routeDefinition`, I would change it, if @vojta-horanek agrees 🙂

Reply:
I do 😄